compliance_checker/ioos.py

```diff
@@ -442,6 +445,15 @@ def __init__(self):
             'references',
         ]
 
+    def setup(self, ds):
+        self.platform_vars = self._find_platform_vars(ds)
+
+    def _find_platform_vars(self, ds):
+        plat_vars = ds.get_variables_by_attributes(platform=
+                                                   lambda p: isinstance(p, str))
+        return {ds.variables[var.platform] for var in plat_vars if
+                var.platform in ds.variables}
```

Comment:
Could `self._platform_vars` be reused here as well?
https://github.com/daltonkell/compliance-checker/blob/0cb51efade62b729af35828f7ba8d6b46d36a603/compliance_checker/ioos.py#L548-L554

Reply:
Maybe, but this check does something different.  It gets the value from each platform attribute from variables and adds it to a set.  The check you wrote doesn't check whether the variable exists in the dataset.  It also adds the global platform attribute to the set, which it shouldn't, as it isn't used to reference variables.